packages/env/lib/download-source.js

```diff
@@ -96,3 +110,46 @@ async function downloadGitSource( source, { onProgress, spinner, debug } ) {
 
 	onProgress( 1 );
 }
+
+/**
+ * Downloads and extracts the zip file at `source.url` into `source.path`.
+ *
+ * @param {Source}   source             The source to download.
+ * @param {Object}   options
+ * @param {Function} options.onProgress A function called with download progress. Will be invoked with one argument: a number that ranges from 0 to 1 which indicates current download progress for this source.
+ * @param {Object}   options.spinner    A CLI spinner which indicates progress.
+ * @param {boolean}  options.debug      True if debug mode is enabled.
+ */
+async function downloadZipSource( source, { onProgress, spinner, debug } ) {
+	const log = debug
+		? ( message ) => {
+				spinner.info( `NodeGit: ${ message }` );
+				spinner.start();
+		  }
+		: () => {};
+	onProgress( 0 );
+
+	log( 'Downloading zip file.' );
+	const zipName = `${ source.path }.zip`;
+	const zipFile = fs.createWriteStream( zipName );
+	await finished(
+		requestProgress( request( source.url ) )
+			.on( 'progress', ( { percent } ) => onProgress( percent ) )
+			.pipe( zipFile )
+	);
+
+	log( 'Extracting to temporary folder.' );
+	const dirName = `${ source.path }.temp`;
+	await extractZip( zipName, { dir: dirName } );
+
+	log( 'Copying to mounted folder and cleaning up.' );
+	await Promise.all( [
+		rimraf( zipName ),
+		...( await fs.promises.readdir( dirName ) ).map( ( file ) =>
```

Comment:
In the other parts of `wp-env` we `const { ... } = require( 'fs' ).promises;` at the top of the file so that there's no need to keep repeating `fs.promises`. Might be worth doing that here for consistency's sake.

Reply:
We also need `createWriteStream` here.